### ISSUE TITLE
axis (and/or axes) specification for softmax

### ISSUE THREAD

**Issue by sergeant-wizard** [pfnet/chainer#2536]
Currently `functions.softmax` only calculates [w.r.t. the second axis](http://docs.chainer.org/en/latest/_modules/chainer/functions/activation/softmax.html),
but I came across a situation where I need the calculation along other axes (specifically, spatial softmax along the pixel direction).
Although swapping axes does work as a workaround, it would be useful to support an optional argument, like so:

```py
F.softmax(x, axes=(2, 3))
```

Just FYI, Tensorflow [has](https://www.tensorflow.org/api_docs/python/tf/nn/softmax)  an option for such requirements.
